src/stdlib_linalg.fypp

```diff
@@ -1065,6 +1067,189 @@ module stdlib_linalg
+     !! input array \( A \), that can have any rank. For generic rank-n arrays, the scalar norm over 
+     !! the whole array is returned by default. If `n>=2` and the optional input dimension `dim` is 
+     !! specified, a rank `n-1` array is returned with dimension `dim` collapsed, containing all 1D 
+     !! array norms evaluated along dimension `dim` only.
+     !! 
+     !!
+     !!### Description
+     !! 
+     !! This `pure subroutine `interface provides methods for computing the vector norm(s) of an array.  
+     !! Supported data types include `real` and `complex`. 
+     !! Input arrays may have generic rank from 1 to ${MAXRANK}$.
+     !!
+     !! Norm type input is mandatory, and it is provided via the `order` argument. 
+     !! This can be provided as either an `integer` value or a `character` string. 
+     !! Allowed metrics are: 
+     !! - 1-norm \( \sum_i{ \left|a_i\right| } \): `order` = 1 or '1'    
+     !! - Euclidean norm \( \sqrt{\sum_i{ a_i^2 }} \): `order` = 2 or '2'
+     !! - p-norm \( \left( \sum_i{ \left|a_i\right|^p }\right) ^{1/p} \): `integer` `order`, order>=3
+     !! - Infinity norm \( \max_i{ \left|a_i\right| } \): order = huge(0) or 'inf'
+     !! - Minus-infinity norm \( \min_i{ \left|a_i\right| } \): order = -huge(0) or '-inf'
+     !!     
+     !!### Example
+     !!
+     !!```fortran
+     !!
+     !!    real(sp) :: a(3,3), na, rown(3)
+     !!    type(linalg_state_type) :: err
+     !!    a = reshape([1, 2, 3, 4, 5, 6, 7, 8, 9], [3, 3])
+     !!
+     !!    ! L2 norm: whole matrix
+     !!    call get_norm(a, na, 2)
+     !!   
+     !!    ! Infinity norms of each row, with error control
+     !!    call get_norm(a, rown, 'inf', dim=2, err=err)     
+     !!     
+     !!```     
+     !!     
+     #:for rk,rt,ri in RC_KINDS_TYPES
+     #:for it,ii in NORM_INPUT_OPTIONS
+        !> Scalar norms: ${rt}$
+        #:for rank in range(1, MAXRANK + 1)                    
+        pure module subroutine norm_${rank}$D_${ii}$_${ri}$(a, nrm, order, err)
+           !> Input ${rank}$-d matrix a${ranksuffix(rank)}$
+           ${rt}$, intent(in) :: a${ranksuffix(rank)}$
+           !> Norm of the matrix.
+           real(${rk}$), intent(out) :: nrm
+           !> Order of the matrix norm being computed.
+           ${it}$, intent(in) :: order
+           !> [optional] state return flag. On error if not requested, the code will stop
+           type(linalg_state_type), intent(out), optional :: err   
+        end subroutine norm_${rank}$D_${ii}$_${ri}$
+        #:endfor
+        !> Array norms: ${rt}$
+        #:for rank in range(2, MAXRANK + 1)
+        pure module subroutine norm_${rank}$D_to_${rank-1}$D_${ii}$_${ri}$(a, nrm, order, dim, err)
+           !> Input matrix a[..]
+           ${rt}$, intent(in), target :: a${ranksuffix(rank)}$
+           !> Dimension to collapse by computing the norm w.r.t other dimensions
+           !  (dim must be defined before it is used for `nrm`)
```

Comment:
What do you mean?

Reply:
[aa734de](https://github.com/fortran-lang/stdlib/pull/871/commits/aa734de2707b808ccfcc2fc8a390c63543d37d5f)